alerter.c

```diff
@@ -1,31 +1,84 @@
 #include <stdio.h>
 #include <assert.h>
 
+
+#define PRODUCTION_ENV  1

```

Comment:
Production code should not be aware of test code or the envt that it is executing in. Given any input it should just process and output or provide certain behavior.  

Abstractions should be provided wither by using interfaces or function pointers or delegates.